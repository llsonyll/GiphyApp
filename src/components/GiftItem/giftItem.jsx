import PropTypes from "prop-types";
import "./giftItem.css";

const GiftItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p> {title} </p>
    </div>
  );
};

GiftItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GiftItem;
