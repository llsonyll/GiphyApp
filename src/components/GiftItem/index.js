export { default } from "./giftItem.jsx";
