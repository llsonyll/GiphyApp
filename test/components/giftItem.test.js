import { render, screen } from "@testing-library/react";
import GiftItem from "../../src/components/GiftItem/giftItem";

describe("Pruebas del GIFT-ITEM component", () => {
  const title = "Titulo";
  const url = "https://punch.com/saitama.jpg";

  test("should match snapshot", () => {
    const { container } = render(<GiftItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should show img with url and alt assigned", () => {
    render(<GiftItem title={title} url={url} />);
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(title);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("should render components title", () => {
    render(<GiftItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
