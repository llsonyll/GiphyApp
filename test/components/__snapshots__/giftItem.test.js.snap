// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del GIFT-ITEM component should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo"
      src="https://punch.com/saitama.jpg"
    />
    <p>
       
      Titulo
       
    </p>
  </div>
</div>
`;
